deepchecks/tabular/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,328 @@
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind)
+        dataset.assert_features()
+        dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
+        predictions = context.model.predict(dataset.features_columns)
+        y_proba = context.model.predict_proba(dataset.features_columns) if \
+            context.task_type in [TaskType.MULTICLASS, TaskType.BINARY] else None
+
+        if self.loss_per_sample is not None:
+            loss_per_sample = self.loss_per_sample[list(dataset.data.index)]
+        else:
+            loss_per_sample = calculate_per_sample_loss(context.model, context.task_type, dataset,
+                                                        self.classes_index_order)
+        dataset = dataset.select(self.columns, self.ignore_columns, keep_label=True)
+        if len(dataset.features) < 2:
+            raise DeepchecksNotSupportedError('Check requires data to have at least two features in order to run.')
+        encoded_dataset = self._target_encode_categorical_features_fill_na(dataset)
+        dummy_model = _DummyModel(test=encoded_dataset, y_pred_test=predictions, y_proba_test=y_proba,
+                                  validate_data_on_predict=False)
+
+        if context.features_importance is not None:
+            feature_rank = context.features_importance.sort_values(ascending=False).keys()
+            feature_rank = np.asarray([col for col in feature_rank if col in encoded_dataset.features])
+        else:
+            feature_rank = np.asarray(encoded_dataset.features)
+
+        scorer = context.get_single_scorer(self.user_scorer)
+        weak_segments = self._weak_segments_search(dummy_model, encoded_dataset, feature_rank,
+                                                   loss_per_sample, scorer)
+        if len(weak_segments) == 0:
+            raise DeepchecksProcessError('Unable to train a error model to find weak segments.')
+        avg_score = round(scorer(dummy_model, encoded_dataset), 3)
+
+        display = self._create_heatmap_display(dummy_model, encoded_dataset, weak_segments, avg_score,
+                                               scorer) if context.with_display else []
+
+        for idx, segment in weak_segments.copy().iterrows():
+            if segment['Feature1'] in encoded_dataset.cat_features:
+                weak_segments['Feature1 range'][idx] = \
+                    self._format_partition_vec_for_display(segment['Feature1 range'], segment['Feature1'], None)[0]
+            if segment['Feature2'] in encoded_dataset.cat_features:
+                weak_segments['Feature2 range'][idx] = \
+                    self._format_partition_vec_for_display(segment['Feature2 range'], segment['Feature2'], None)[0]
+
+        display_msg = 'Showcasing intersections of features with weakest detected segments.<br>Can ' \
+                      'observe the full list of weak segments in the check result value. '
```

Comment:
Instead of "can observe", which makes this an incomplete sentence, maybe something like "The full list of weak segments can be observed in the check result value"?